src/jmcomic/jm_plugin.py

```
@@ -111,7 +111,7 @@ def wait_until_finish(self):
     def decide_filepath(self,
                         album: Optional[JmAlbumDetail],
                         photo: Optional[JmPhotoDetail],
-                        filename_rule: str, suffix: str, base_dir: Optional[str],
+                        filename_rule: Optional[str], suffix: Optional[str], base_dir: Optional[str],
```

Ruff (0.13.1)

`Optional` may be undefined, or defined from star imports

(F405)